packages/ui/src/components/_utils/analytics.ts

```diff
@@ -4,21 +4,34 @@ import React from 'react'
 
 import { version } from '@open-condo/ui/package.json'
 
+interface IAnalyticsProps<K> {
+    location: string
+    component: K
+}
+
 type ComponentSpecificClickEventProps = {
     Banner: { title: string }
     Button: { value: string, type: string }
     'Typography.Link': { value: string, href?: string }
 }
 
-type CommonAnalyticsProps<K> = {
```

Comment:
why not just:
```typescript
type AnalyticEvent = 'click' | 'check'
type CommonAnalyticsProps<Event extends AnalyticEvent, K> = {
     event: Event,
     location: string,
     component: K
}
```

Reply:
K can also extends `union` of `keyof`

Reply:
> K can also extends `union` of `keyof`

What is the reason of extending `AnalyticsEvent` at your example?